server/src/test/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerServiceTests.java

```diff
@@ -817,4 +817,30 @@ public void testCustomCircuitBreakers() {
     private static long mb(long size) {
         return new ByteSizeValue(size, ByteSizeUnit.MB).getBytes();
     }
+
+    public void testUpdatingUseRealMemory() {
+        try (
+            HierarchyCircuitBreakerService service = new HierarchyCircuitBreakerService(
+                Settings.EMPTY,
+                Collections.emptyList(),
+                new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS)
+            )
+        ) {
+            // use real memory default true
+            assertTrue(service.isTrackRealMemoryUsage());
```

Comment:
Hi @ViggoC, I think we need to bring back the unit tests that ensure that changing USE_REAL_MEMORY_USAGE_SETTING affects TOTAL_CIRCUIT_BREAKER_LIMIT_SETTING. Rather than calling `service.updateUseRealMemorySetting` directly, would it be possible to use a local clusterSettings, like in `testAllocationBucketsBreaker` and then invoke `clusterSettings.applySettings` on it, which will trigger the updates to both properties? You can find examples of this being done in our `IndexShardTests` unit tests.

Since these two settings are dependent, we have to have unit tests in place to ensure the updates are done in tandem.

You can keep the existing code as-is, perhaps we can just add the updating through the applySettings method and ensure the properties change for total.

Reply:
fix on 694093e. Thank you again for your patience. I have been troubled by how to trigger settings updates in unit tests, and your guidance has helped me a lot.

Reply:
Great, thank you!